### PR TITLE
[web][SIMS #421]Education Program Offering WEB

### PR DESCRIPTION
**Pull request by guru-aot**
Education Program Offering 

- Form integration
- Web Routes API creation

### CODE REVIEW

File: sources/packages/web/src/router/InstitutionRoutes.ts [hunk 1]
```diff
@@ -75,6 +76,15 @@ export const institutionRoutes: Array<RouteRecordRaw> = [
             },
             props: true,
           },
+          {
+            path: `${AppRoutes.LocationOfferings}/:locationId/education-program/:programId`,
```
Comment (ann-aot) [inline sources/packages/web/src/router/InstitutionRoutes.ts:80]: all strings can be registered in `/AppRoutes.ts`, here `education-program`

Comment (ann-aot) [inline sources/packages/web/src/router/InstitutionRoutes.ts:80]: we can register the full URL in `AppRoutes.ts`

File: sources/packages/web/src/services/EducationProgramOfferingService.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  private static instance: EducationProgramOfferingService;
+
+  public static get shared(): EducationProgramOfferingService {
+    return this.instance || (this.instance = new this());
+  }
+
+  public async createProgramOffering(
+    locationId: number,
+    programId: number,
+    data: any,
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/services/EducationProgramOfferingService.ts:14]: Try to make the type explicit here. My previous PR was missing it due to an issue with Sonar Cloud.

File: sources/packages/web/src/services/http/EducationProgramOfferingApi.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import HttpBaseClient from "./common/HttpBaseClient";
+
+export class EducationProgramOfferingApi extends HttpBaseClient {
+  public async createProgramOffering(
+    locationId: number,
+    programId: number,
+    createProgramOfferingDto: any,
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/services/http/EducationProgramOfferingApi.ts:7]: Please try to make the type explicit here.

**Review by andrewsignori-aot — APPROVED** [review]
Please take a look at the comments. Besides that looks good to me 👍

File: sources/packages/web/src/router/InstitutionRoutes.ts (hunk 1 above)
Comment (guru-aot) [reply sources/packages/web/src/router/InstitutionRoutes.ts]: Done

File: sources/packages/web/src/services/EducationProgramOfferingService.ts (hunk 2 above)
Comment (guru-aot) [reply sources/packages/web/src/services/EducationProgramOfferingService.ts]: Done created a contract

File: sources/packages/web/src/services/http/EducationProgramOfferingApi.ts (hunk 3 above)
Comment (guru-aot) [reply sources/packages/web/src/services/http/EducationProgramOfferingApi.ts]: Done created a contract

**Review by ann-aot — APPROVED** [review]
LGTM

**Pull request merged**